db/migrate/20230405121453_add_password_algorithm_to_users.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddPasswordAlgorithmToUsers < ActiveRecord::Migration[7.0]
+  def up
+    add_column :users, :password_algorithm, :string, limit: 64
+    execute <<~SQL
+      UPDATE users SET password_algorithm = '$pbkdf2-sha256$i=64000,l=32$'
+      WHERE users.password_hash IS NOT NULL
+    SQL
```

Comment:
This will end up locking most if not all rows on the users table within the same transaction. Perhaps run this backfill in batches to avoid any potential locking issue that might arise on sites with a larger `users` table?